railties/lib/rails/code_statistics.rb

```diff
@@ -9,6 +9,8 @@ class CodeStatistics #:nodoc:
                 'Job tests',
                 'Integration tests']
 
+  DEFAULT_PATTERN = /^(?!\.).*?\.(rb|js|coffee|rake)$/
```

Comment:
No need for extraction. Ruby deduplicates it when parsing the file the first time.